game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/RollNormal.java

```diff
@@ -0,0 +1,30 @@
+package games.strategy.triplea.delegate.battle.steps.fire;
+
+import static games.strategy.triplea.delegate.battle.BattleState.Side.DEFENSE;
+import static games.strategy.triplea.delegate.battle.BattleState.UnitBattleFilter.ACTIVE;
+
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.DiceRoll;
+import java.util.function.BiFunction;
+
+/** Rolls dice for normal (basically, anything that isn't AA) dice requests */
+public class RollNormal implements BiFunction<IDelegateBridge, RollDice, DiceRoll> {
+
+  @Override
+  public DiceRoll apply(final IDelegateBridge bridge, final RollDice step) {
+    return DiceRoll.rollDice(
```

Comment:
side-question, any plans to refactor the rollDice API in the future? Seemingly we should be able to pass a step and bridge to it and little else. It's a bit surprising there is enough data between the 2 or 3 variables to supply the nearly dozen parameters.

Reply:
Yes, I do plan to refactor it.  I'd actually like to just have one `rollDice` api instead of a "normal" and an "aa" roll dice api.

The refactoring for `TotalPowerAndTotalRolls` is the start of that work.  The list of parameters should shrink as I group them into more related objects.